test/luatest_helpers/cluster.lua

```diff
@@ -72,18 +72,41 @@ function Cluster:stop()
 end
```

Comment:
I think the patch isn't really needed. Because in a test you either should not rely on who is the master (boostrap leader) and should find it dynamically, or if you really want one node to be the master - you can give it `read_only = false` and to the others - `read_only = true`. You could do that here too by the way. Probably it could be much simpler and would work faster because you can start all nodes at once. Also you can use non-random UUIDs - give minimal UUID to the node which needs to be the bootstrap master (voting for boot leader uses UUID comparison). It does not need to be sequential if you want just to have a specific node as a master.

This sequential startup I am sure won't really fix the flakyness. At least I can't remember when not knowing exact boot master was the reason for a fail. Usually fail happens after bootstrap in something not related to that. But I do believe this thing makes tests notably slower which is quite bad, and while in theory it could make tests probably more stable, it will come at a cost of smaller coverage.

And finally the API looks a bit strange. You assume the master is the first instance. I would rather expect it to be more explicit. Like a flag for the server which you want to be the master.

Reply:
Thanks for in depth review! Setting only one server in cluster to rw and others to ro does the trick with less effort and faster! It doesn't make the test completely reproducible, but makes it fail a lot more often which helps debugging the real fix. Closing this PR now.